### ISSUE TITLE
Rule S1121: FP when `IsPattern` syntax is used

### ISSUE THREAD

**Issue by TrevorPilley** [SonarSource/sonar-dotnet#4264]
### Description

S1121 is fine with `!= null` but raises a warning for `is not null` with an inline assignment whilst reading lines from a `StreamReader`.

### Repro steps

No build warning:

```csharp
string line?
while ((line = await reader.ReadLineAsync()) != null)
{
}
```

However, if I change `!=` to `is not`:

```csharp
string line?
while ((line = await reader.ReadLineAsync()) is not null)
{
}
```

I get the following build warning:

> warning S1121: Extract the assignment of ‘line’ from this expression

### Expected behavior

No S1121 warning.

### Actual behavior

An S1121 warning flagged by the compiler.

### Known workarounds

Please provide a description of any known workarounds.

### Related information

Building with C#9 language, net5.0 target framework and SonarAnalyzer.CSharp 8.21.0.30542 in VS Code 1.55.2.

* C#/VB.NET Plugins version
* Visual Studio version
* MSBuild / dotnet version 
* SonarScanner for .NET version (if used)
* Operating System

**Comment by costin-zaharia-sonarsource** [comment]
Hi @TrevorPilley, I can confirm the false positive. Thanks for your feedback.

**Comment by TrevorPilley** [comment]
Hi, just wondering if there's an expected release date for this fix?

**Comment by costin-zaharia-sonarsource** [comment]
Hi @TrevorPilley, your reminder arrived at the right time since we are currently doing a short hardening sprint. I expect that the fix for this to be part of the next release which should happen next Monday (November 15).
